src/pdf_beamtime/src/pdf_beamtime_fidpose_server.cpp

```diff
@@ -0,0 +1,626 @@
+
+    case State::PICKUP_RETREAT:
+    case State::PLACE_APPROACH:
+    case State::PLACE:
+      return_sample();
+      break;
+
+    case State::PICKUP:
+    case State::GRASP_FAILURE: {
+        // Move back to pickup approach
+        std::pair<double, double> new_wrist_angles = tf_utilities_->get_wrist_elbow_alignment(
+          move_group_interface_);
+        adjusted_pickup_ = goal->pickup_approach;
+        adjusted_pickup_[4] = new_wrist_angles.first;
+        adjusted_pickup_[5] = new_wrist_angles.second;
+        inner_state_machine_->move_robot(
+          move_group_interface_,
+          adjusted_pickup_);
+      }
+      break;
+
+    case State::RELEASE_FAILURE:
+    case State::RELEASE_SUCCESS: {
+        std::pair<double, double> new_wrist_angles = tf_utilities_->get_wrist_elbow_alignment(
+          move_group_interface_);
+        adjusted_place_ = goal->place_approach;
+        adjusted_place_[4] = new_wrist_angles.first;
+        adjusted_place_[5] = new_wrist_angles.second;
+        inner_state_machine_->move_robot(
+          move_group_interface_,
+          adjusted_place_);
+      }
+      break;
+
+    default:
+      break;
+  }
+  reset_fsm();
+  inner_state_machine_->set_internal_state(Internal_State::RESTING);
+  RCLCPP_INFO(node_->get_logger(), "Cleanup is complete");
+}
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+
+  // Create a ROS logger for main scope
+  auto const logger = rclcpp::get_logger("hello_moveit");
+  using namespace std::chrono_literals;
+
+  // Create a node for synchronously grabbing params
+  auto parameter_client_node = rclcpp::Node::make_shared("param_client");
+  auto parent_parameters_client =
+    std::make_shared<rclcpp::SyncParametersClient>(parameter_client_node, "move_group");
+  // Boiler plate wait block
+  while (!parent_parameters_client->wait_for_service(1s)) {
+    if (!rclcpp::ok()) {
+      RCLCPP_ERROR(
+        logger, "Interrupted while waiting for the service. Exiting.");
+      return 0;
+    }
+    RCLCPP_INFO(logger, "move_group service not available, waiting again...");
+  }
+  // Get robot config parameters from parameter server
+  auto parameters = parent_parameters_client->get_parameters(
+    {"robot_description_semantic",
+      "robot_description"});
+
+  // Set node parameters using NodeOptions
+  rclcpp::NodeOptions node_options;
+  node_options.automatically_declare_parameters_from_overrides(true);
+  node_options.parameter_overrides(
+  {
+    {"robot_description_semantic", parameters[0].value_to_string()},
+    {"robot_description", parameters[1].value_to_string()}
+  });
+
+  rclcpp::executors::MultiThreadedExecutor executor;
+
+  auto beamtime_server = std::make_shared<PdfBeamtimeFidPoseServer>(
+    "ur_arm",
+    node_options);
+
+  executor.add_node(beamtime_server->getNodeBaseInterface());
+  executor.add_node(beamtime_server->getInterruptNodeBaseInterface());
+  executor.spin();
+
+  rclcpp::shutdown();
+  return 0;
+}
```

Comment:
Why an entry point for the original server, but not an entry point here? Happy for either, but consistency would help.

Reply:
A separate entry point is added:
https://github.com/maffettone/erobs/blob/a54c58888c733d986f5fe443e0b3d04d33e9c634/src/pdf_beamtime/src/beamtime_fidpose_entrypoint.cpp